src/bin/set-version/main.rs

```diff
@@ -0,0 +1,200 @@
+    bufwtr
+        .print(&buffer)
+        .chain_err(|| "Failed to print dry run message")
+}
+
+impl Manifests {
+    /// Get all manifests in the workspace.
+    fn get_all(manifest_path: &Option<PathBuf>) -> Result<Self> {
+        let mut cmd = cargo_metadata::MetadataCommand::new();
+        cmd.no_deps();
+        if let Some(path) = manifest_path {
+            cmd.manifest_path(path);
+        }
+        let result = cmd.exec().map_err(|e| {
+            Error::from(e.compat()).chain_err(|| "Failed to get workspace metadata")
+        })?;
+        result
+            .packages
+            .into_iter()
+            .map(|package| {
+                Ok((
+                    LocalManifest::try_new(Path::new(&package.manifest_path))?,
+                    package,
+                ))
+            })
+            .collect::<Result<Vec<_>>>()
+            .map(Manifests)
+    }
+
+    /// Get the manifest specified by the manifest path. Try to make an educated guess if no path is
+    /// provided.
+    fn get_local_one(manifest_path: &Option<PathBuf>) -> Result<Self> {
+        let resolved_manifest_path: String = find(&manifest_path)?.to_string_lossy().into();
+
+        let manifest = LocalManifest::find(&manifest_path)?;
+
+        let mut cmd = cargo_metadata::MetadataCommand::new();
+        cmd.no_deps();
+        if let Some(path) = manifest_path {
+            cmd.manifest_path(path);
+        }
+        let result = cmd
+            .exec()
+            .map_err(|e| Error::from(e.compat()).chain_err(|| "Invalid manifest"))?;
+        let packages = result.packages;
+        let package = packages
+            .iter()
+            .find(|p| p.manifest_path.to_string_lossy() == resolved_manifest_path)
+            // If we have successfully got metadata, but our manifest path does not correspond to a
+            // package, we must have been called against a virtual manifest.
+            .chain_err(|| {
+                "Found virtual manifest, but this command requires running against an \
+                 actual package in this workspace. Try adding `--workspace`."
+            })?;
+
+        Ok(Manifests(vec![(manifest, package.to_owned())]))
+    }
+
+    /// Update the manifests on disk with the given version.
+    fn set_version(self, dry_run: bool, version: Version) -> Result<()> {
+        if dry_run {
+            dry_run_message()?;
+        }
+        for (mut manifest, package) in self.0 {
+            let bufwtr = BufferWriter::stdout(ColorChoice::Always);
+            let mut buffer = bufwtr.buffer();
+            writeln!(
+                &mut buffer,
+                "{}: {} -> {}",
+                package.name, package.version, version
+            )
+            .chain_err(|| "Failed to write dry run message")?;
+            bufwtr
+                .print(&buffer)
+                .chain_err(|| "Failed to print dry run message")?;
+            if !dry_run {
```

Comment:
Will change this to always run, even if dry run and update copy-pasted error messages